torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -602,6 +602,19 @@ def _collect_local_shard(module: torch.nn.Module) -> torch.nn.Module:
     """
     Hook a module with local shards collection in the forward pass.
 
+    When the local tensor only has one dimension, we increase one more dimension
+    for reshard. We then do squeeze here manually to reduce the dimension.
```

Comment:
I don't think we should specify reshard or sharded linear in the docs of this method. I think the docs should be something like this:

```
This API is typically used to convert a sharded representation back to data parallel representation. In particular, it returns the local tensor for this Shard. If the size along the sharding dimension for the local tensor is 1, this dimension is removed from the final result. For example a [4, 16] ShardedTensor across 4 ranks is typically a local Tensor of size [16] across each rank and not [1, 16] across each rank.
```

Reply:
sure.